eth/transactionManager.go

```diff
@@ -0,0 +1,233 @@
+	"context"
+	"fmt"
+	"math/big"
+	"strings"
+	"sync"
+	"time"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/golang/glog"
+	"github.com/livepeer/go-livepeer/common"
+)
+
+type TransactionManager struct {
+	txTimeout time.Duration
+	replaceTx bool
+
+	queue transactionQueue
+
+	// subscriptions
+	feed  event.Feed
+	scope event.SubscriptionScope
+
+	eth *ethclient.Client
+	gpm *GasPriceMonitor
+	am  AccountManager
+
+	cond sync.Cond
+
+	quit chan struct{}
+}
+
+type transactionQueue []*types.Transaction
+
+func (tq *transactionQueue) add(tx *types.Transaction) {
+	*tq = append(*tq, tx)
+}
+
+func (tq *transactionQueue) pop() *types.Transaction {
+	if tq.length() == 0 {
+		return nil
+	}
+	tx := (*tq)[0]
+	*tq = (*tq)[1:]
+	return tx
+}
+
+func (tq transactionQueue) length() int {
+	return len(tq)
+}
+
+func (tq transactionQueue) peek() *types.Transaction {
+	if tq.length() == 0 {
+		return nil
+	}
+	return tq[0]
+}
+
+func NewTransactionManager(eth *ethclient.Client, gpm *GasPriceMonitor, am AccountManager, txTimeout time.Duration, replace bool) *TransactionManager {
+	return &TransactionManager{
+		txTimeout: txTimeout,
+		replaceTx: replace,
+		eth:       eth,
+		gpm:       gpm,
+		am:        am,
+		queue:     transactionQueue{},
+		quit:      make(chan struct{}),
+	}
+}
+
+func (tm *TransactionManager) SendTransaction(ctx context.Context, tx *types.Transaction) error {
+	sendErr := tm.eth.SendTransaction(ctx, tx)
+
+	txLog, err := newTxLog(tx)
+	if err != nil {
+		txLog.method = "unknown"
+	}
+
+	if sendErr != nil {
+		glog.Infof("\n%vEth Transaction%v\n\nInvoking transaction: \"%v\". Inputs: \"%v\"   \nTransaction Failed: %v\n\n%v\n", strings.Repeat("*", 30), strings.Repeat("*", 30), txLog.method, txLog.inputs, sendErr, strings.Repeat("*", 75))
+		return sendErr
+	}
+
+	// Add transaction to queue
+	tm.cond.L.Lock()
+	tm.queue.add(tx)
+	tm.cond.L.Unlock()
+	tm.cond.Signal()
+
+	glog.Infof("\n%vEth Transaction%v\n\nInvoking transaction: \"%v\". Inputs: \"%v\"  Hash: \"%v\". \n\n%v\n", strings.Repeat("*", 30), strings.Repeat("*", 30), txLog.method, txLog.inputs, tx.Hash().String(), strings.Repeat("*", 75))
+
+	return nil
+}
+
+func (tm *TransactionManager) Subscribe(sink chan<- *types.Receipt) event.Subscription {
+	return tm.scope.Track(tm.feed.Subscribe(sink))
+}
+
+func (tm *TransactionManager) Start() {
+	tm.checkTxLoop()
+}
+
+func (tm *TransactionManager) Stop() {
+	defer tm.cond.L.Unlock()
+	tm.scope.Close()
+	close(tm.quit)
+}
+
+func (tm *TransactionManager) wait(tx *types.Transaction) error {
+	ctx, cancel := context.WithTimeout(context.Background(), tm.txTimeout)
+	defer cancel()
+
+	receipt, err := bind.WaitMined(ctx, tm.eth, tx)
+	if err != nil {
+		return err
+	}
+
+	tm.feed.Send(receipt)
+
+	if receipt.Status == uint64(0) {
```

Comment:
What about just letting the caller i.e. CheckTx() handle this check? Otherwise, we currently have this check here and in CheckTx().